clang/include/clang/Basic/Attr.td

```diff
@@ -1393,7 +1393,7 @@ def IntelReqdSubGroupSize: InheritableAttr {
   let Spellings = [GNU<"intel_reqd_sub_group_size">,
                    CXX11<"intel", "reqd_sub_group_size">];
   let Args = [ExprArgument<"Value">];
-  let Subjects = SubjectList<[Function, CXXMethod], ErrorDiag>;
```

Comment:
This change wasn't strictly necessary in this patch, but I added it as a drive-by fix because I was already touching this attribute. It's an effectively NFC change (the pragma subject test had to be updated but the functionality is identical).

You don't need to add `CXXMethod` to the list because `CXXMethodDecl` already inherits from `FunctionDecl`, so the existing `Function` subject covers both cases.